lte/gateway/c/oai/lib/pipelined_client/PipelinedClientAPI.cpp

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PipelinedClientAPI.h"
+
+#include <grpcpp/security/credentials.h>
+#include <cstdint>
+#include <cstring>
+#include <string>
+
+#include "conversions.h"
+#include "common_defs.h"
+#include "service303.h"
+#include "spgw_types.h"
+#include "common_types.h"
+
+#include "PipelinedServiceClient.h"
+
+using grpc::Channel;
+using grpc::ChannelCredentials;
+using grpc::CreateChannel;
+using grpc::InsecureChannelCredentials;
+using grpc::Status;
+using magma::lte::CauseIE;
+using magma::lte::PipelinedServiceClient;
+using magma::lte::UESessionContextResponse;
+
+void handle_upf_classifier_rpc_call_done(
+    const grpc::Status& status, UESessionContextResponse response);
+
+int upf_classifier_add_tunnel(
+    struct in_addr ue, struct in6_addr* ue_ipv6, int vlan, struct in_addr enb,
+    uint32_t i_tei, uint32_t o_tei, const char* imsi,
+    struct ip_flow_dl* flow_dl, uint32_t flow_precedence_dl, const char* apn) {
+  std::string imsi_str = std::string(imsi);
+  std::string apn_str  = std::string(apn);
+
+  if (NULL == ue_ipv6) {
```

Comment:
For all of these I think it's cleaner to read `ue_ipv6 == NULL` not `NULL == ue_ipv6`

Reply:
Comments addressed.